Lib/test/test_inspect.py

```diff
@@ -690,15 +691,27 @@ def test_nested_class_definition(self):
         self.assertSourceEqual(mod2.cls183.cls185, 185, 188)
 
     def test_class_decorator(self):
-        self.assertSourceEqual(mod2.cls197, 195, 202)
-        self.assertSourceEqual(mod2.cls197.cls201, 199, 202)
+        self.assertSourceEqual(mod2.cls196, 194, 201)
+        self.assertSourceEqual(mod2.cls196.cls200, 198, 201)
 
     def test_multiple_children_classes(self):
-        self.assertSourceEqual(mod2.cls204, 204, 210)
-        self.assertSourceEqual(mod2.cls204.cls205, 205, 207)
-        self.assertSourceEqual(mod2.cls204.cls205.cls206, 206, 207)
-        self.assertSourceEqual(mod2.cls204.cls208, 208, 210)
-        self.assertSourceEqual(mod2.cls204.cls208.cls209, 209, 210)
+        self.assertSourceEqual(mod2.cls203, 203, 209)
+        self.assertSourceEqual(mod2.cls203.cls204, 204, 206)
+        self.assertSourceEqual(mod2.cls203.cls204.cls205, 205, 206)
+        self.assertSourceEqual(mod2.cls203.cls207, 207, 209)
+        self.assertSourceEqual(mod2.cls203.cls207.cls208, 208, 209)
+
+    def test_nested_class_definition_inside_function(self):
+        self.assertSourceEqual(mod2.func212(), 213, 215)
+        self.assertSourceEqual(mod2.cls213, 219, 223)
+        self.assertSourceEqual(mod2.cls213().func220(), 221, 222)
+
+    def test_nested_class_definition_inside_async_function(self):
+        import asyncio
+        self.assertSourceEqual(asyncio.run(mod2.func226()), 227, 229)
+        self.assertSourceEqual(mod2.cls227, 233, 237)
+        self.assertSourceEqual(asyncio.run(mod2.cls227().func234()), 235, 236)
+        asyncio.set_event_loop_policy(None)
```

Comment:
Use `addCleanup()`.

Reply:
Seems this test class inherits from `unittest.Test` and `self.addCleanup(asyncio.set_event_loop_policy, None)` doesn't work. Should I use `asyncio.set_event_loop_policy(None)` or change the base class from which it's inherited?

Reply:
Sorry, it works. I think I was having test failures and tested it during that time causing confusion that it's an environment change. Added the change as part of c6e5e2c133069bcb45568b141636b6b78e2348d3 and verified locally.

```
./python.exe -m test --fail-env-changed test_inspect
Run tests sequentially
0:00:00 load avg: 2.57 [1/1] test_inspect

== Tests result: SUCCESS ==

1 test OK.

Total duration: 2 sec 345 ms
Tests result: SUCCESS
```